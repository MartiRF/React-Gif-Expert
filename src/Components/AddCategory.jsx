import React, { useState } from 'react'
import PropsTypes from 'prop-types'
export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('')

    const onSumit = (e) => {
        e.preventDefault()
        if(inputValue.trim().length <= 1) return;
        // onAddCategori((categories) => [inputValue,...categories]);
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    const onInputChange = ({ target })  => {
        setInputValue(target.value)
    }


  return (
    <form onSubmit={onSumit} aria-label="form">
        <input 
        type="text" 
        placeholder='Buscar Gifts'
        value={inputValue}
        onChange={onInputChange}
        />
    </form>    
)
}

// PropsTypes
AddCategory.propTypes = {
    onNewCategory:PropsTypes.func.isRequired,
}