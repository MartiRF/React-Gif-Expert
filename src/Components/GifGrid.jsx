import { GifItem } from './GifItem'
import { useFechtGifs } from '../hooks/useFechtGifs';
import PropTypes from 'prop-types'
// import React, { useEffect, useState } from 'react'
// import { getGifs } from '../helpers/getGifs'

export const GifGrid = ({ category }) => {
  
  // Custom HOOK
  const { data, isLoading } = useFechtGifs(category);

  return (
    <>
        <h3>{category}</h3>
        {
          isLoading?<h2>Cargando</h2>:null
        }
        
        <div className='card-grid'>
          {data?.map(( image ) => (
            <GifItem 
              key={image.id}
              {...image}
            />
          ))}
        </div>
    </>
  )
}

// PropTypes
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}