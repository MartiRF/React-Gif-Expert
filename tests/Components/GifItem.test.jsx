import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/Components/GifItem'

describe('Prueba de <GifItem />', () => {

    const url = 'http://ejemplo.com/'
    const title = 'titulo ejemplo';

    test('Snapshop', () => {
        const {container} = render( <GifItem url={url} title={title} />)
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el url y el ALT indicado', () => {
        render( <GifItem url={url} title={title} />)
        // screen.debug()
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title)
    });

    test('debe de mostrar el titulo en el componente', () => {
        render( <GifItem url={url} title={title} />)
        expect( screen.getByText(title)).toBeTruthy();
    });
});