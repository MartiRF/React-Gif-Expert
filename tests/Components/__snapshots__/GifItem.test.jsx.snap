// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GifItem /> Snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo ejemplo"
      src="http://ejemplo.com/"
    />
    <p>
      titulo ejemplo
    </p>
  </div>
</div>
`;
